### PR TITLE
primitive-types: scale-info support

### LINKED ISSUE #311: [primitive-types] Implement metadata traits for the primitive_types::U256
I'm coding a smart contract with Parity Ink! in which I have to store an unsigned 256-bit number (u256 not supported by Rust), so I planned to import `primitive-types` by adding:
- `primitive-types = { version = "0.6.2", default-features = false, features = ["codec"] }` in `Cargo.toml` file;
- `use primitive_types::U256;` in `lib.rs` file.
By running `cargo contract build` --> NO ERRORS and `wasm` file generated correctly.
But when running `cargo contract generate-metadata`, I get the following errors:
```
error[E0277]: the trait bound `primitive_types::U256: type_metadata::type_id::HasTypeId` is not satisfied
   --> lib.rs:8:1
    |
8   | #[ink::contract(version = "0.1.0")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `type_metadata::type_id::HasTypeId` is not implemented for `primitive_types::U256`
    | 
   ::: /home/luca/.cargo/git/checkouts/ink-1add513eda8f5a89/8bcf381/abi/src/specs.rs:625:12
    |
625 |         T: Metadata,
    |            -------- required by this bound in `ink_abi::specs::TypeSpec::with_name_segs`
    |
    = note: required because of the requirements on the impl of `type_metadata::Metadata` for `primitive_types::U256`

error[E0277]: the trait bound `primitive_types::U256: type_metadata::type_def::HasTypeDef` is not satisfied
   --> lib.rs:8:1
    |
8   | #[ink::contract(version = "0.1.0")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `type_metadata::type_def::HasTypeDef` is not implemented for `primitive_types::U256`
    | 
   ::: /home/luca/.cargo/git/checkouts/ink-1add513eda8f5a89/8bcf381/abi/src/specs.rs:625:12
    |
625 |         T: Metadata,
    |            -------- required by this bound in `ink_abi::specs::TypeSpec::with_name_segs`
    |
    = note: required because of the requirements on the impl of `type_metadata::Metadata` for `primitive_types::U256`

```
It seems a problem related to lack of implementations of some metadata traits for the `primitive_types::U256` type.
Could someone implement those traits in the `type-metadata` crate or wrap the `primitive_types::U256` type in a tuple-struct and implement the trait on it.

Thank you

### PR DESCRIPTION
**Pull request by ordian**
Fixes #311.

### CODE REVIEW

**Comment by dvdplm** [comment]
@ordian what is the status of this?

**Comment by ordian** [comment]
@dvdplm blocked on https://github.com/type-metadata/type-metadata/issues/41

**Comment by ordian** [comment]
@ascjones should this PR target https://github.com/paritytech/scale-info instead? Is the plan to abandon type-metadata?

**Comment by ascjones** [comment]
@ordian yes. scale-info supersedes type-metadata. I'm working towards a 0.1 release as we speak: https://github.com/paritytech/scale-info/milestone/1

File: primitive-types/Cargo.toml [hunk 1]
```diff
@@ -13,8 +13,8 @@ uint = { version = "0.8.3", path = "../uint", default-features = false }
 impl-serde = { version = "0.3.0", path = "impls/serde", default-features = false, optional = true }
 impl-codec = { version = "0.4.1", path = "impls/codec", default-features = false, optional = true }
 impl-rlp = { version = "0.2", path = "impls/rlp", default-features = false, optional = true }
-# type-metadata = { version = "0.1", features = ["derive"], default-features = false, optional = true }
-type-metadata = { git = "https://github.com/type-metadata/type-metadata", default-features = false, features = ["derive"], optional = true }
+# scale-info = { version = "0.1", features = ["derive"], default-features = false, optional = true }
+scale-info = { git = "https://github.com/paritytech/scale-info", default-features = false, features = ["derive"], optional = true }
```
Comment (Robbepop) [inline primitive-types/Cargo.toml]: Why don't you use the version `0.2` that can be found at [crates.io](crates.io/crates/scale-info)?

Comment (ordian) [reply primitive-types/Cargo.toml]: (I'm back from vacation, sorry for the delay in reply)
this comment is on an outdated commit, this PR already uses 0.2

Comment (Robbepop) [reply primitive-types/Cargo.toml]: Yep I just saw and by now there is version `0.3` on crates.io. :D

**Pull request merged**